### PR TITLE
Fix the zip-slip vulnerability

### LINKED ISSUE #2852: IOUtil unzip doesn't handle correctly file paths
`com.hazelcast.jet.impl.util.IOUtil.unzip` (https://github.com/hazelcast/hazelcast-jet/blob/master/hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/util/IOUtil.java#L122) doesn't sanitize paths within unzipped archive which can lead to creating files outside of target directory.

### PR DESCRIPTION
**Pull request by viliam-durina**
Fixes #2852

### CODE REVIEW

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder` of your PR failed.
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:checkstyle (default) on project hazelcast-jet-core: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There is 1 error reported by Checkstyle 8.29 with /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/util/IOUtilTest.java:128: Line is longer than 123 characters (found 126). [LineLength]
--------------------------
</pre>
</details>

**Comment by hz-highfive** [comment]
The job `jet-oss-pr-builder` of your PR failed.
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-checkstyle-plugin:3.1.0:checkstyle (default) on project hazelcast-jet-core: An error has occurred in Checkstyle report generation. Failed during checkstyle execution: There is 1 error reported by Checkstyle 8.29 with /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/checkstyle/checkstyle.xml ruleset. -> [Help 1]
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/jet-oss-pr-builder/hazelcast-jet-core/src/test/java/com/hazelcast/jet/impl/util/IOUtilTest.java:130:5: '{' at column 5 should be on the previous line. [LeftCurly]
--------------------------
</pre>
</details>

**Review by jbartok — APPROVED** [review]
LGTM

File: extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/ChangeRecord.java [hunk 1]
```diff
@@ -66,7 +66,7 @@
      * INSERT} events in both cases (a notable example is MySQL).
      *
      * @throws ParsingException if the timestamp field isn't present or
-     *                          is unparsable
```
[the remark this answers is not in the document]
Comment (viliam-durina) [reply extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/ChangeRecord.java:69]: Google says the one with "e" is 4 times as common... I don't know...

**Pull request merged**